### PR TITLE
Error boundary and error message

### PR DESCRIPTION
**Pull request by eskil0312**

### CODE REVIEW

File: src/components/general/ErrorMessage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import styles from "./styles.less";
+import * as Icons from "./errorIcons";
+import classNames from "classnames";
+
+export enum ErrorTypes {
+    error= "error",
+    accessDenied= "accessDenied",
+    notFound= "notFound",
+    noData= "noData",
+    noAttachment= "noAttachment",
```
Comment (MartinForre) [inline src/components/general/ErrorMessage/index.tsx]: noAttachment and noActionCreated look like they're specific for meetings/reviews. We shouldn't have app specific error types and messages in a common component. Should be able to use the title and message props to provide that

File: src/components/general/ErrorMessage/styles.less [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    .messageContainer {
+        position: relative;
+        top: 0; left: 0; bottom: 0; right: 0;
+        display: flex;
+        align-content: center;
+        align-items: center;
+        justify-content: center;
+        flex-direction: column;
+        background: rgba(255, 255, 255, .8);
```
Comment (MartinForre) [inline src/components/general/ErrorMessage/styles.less]: var(--colors-something)?

File: src/components/general/ErrorMessage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+type ErrorMessageProps = {
+    hasError?: boolean,
+    errorType?:ErrorTypes,
+    message?: any,
+    resourceName?: string,
+    title?: string,
+    children?: any,
+    icon?: any,
+    button?: string,
```
Comment (MartinForre) [inline src/components/general/ErrorMessage/index.tsx]: Could this be something a bit more descriptive than "button"? "action" or something like that?

File: src/components/general/ErrorBoundary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      }
+
+    state = {
+        hasError: false,
+        retries: 0,
+    };
+
+    componentDidCatch(error, errorInfo) {
+
+        this.setState({ hasError: true });
```
Comment (MartinForre) [inline src/components/general/ErrorBoundary/index.tsx]: Persist the error and errorInfo to provide more info to the error message?

File: src/components/general/ErrorMessage/index.tsx (hunk 1 above)
Comment (eskil0312) [reply src/components/general/ErrorMessage/index.tsx]: Removed noAttachment and noActionCreated

File: src/components/general/ErrorMessage/styles.less (hunk 2 above)
Comment (eskil0312) [reply src/components/general/ErrorMessage/styles.less]: There was no background color was not in the design sheet so it was removed

File: src/components/general/ErrorMessage/index.tsx (hunk 3 above)
Comment (eskil0312) [reply src/components/general/ErrorMessage/index.tsx]: Changed to action

File: src/components/general/ErrorBoundary/index.tsx (hunk 4 above)
Comment (eskil0312) [reply src/components/general/ErrorBoundary/index.tsx]: Now uses the error.message to provide more info. 
Is it necessary to provide more info than the message to the user?
All error information was used to provide to appInsight, but we don't have a implementation for that yet.

**Comment by eskil0312** [comment]
The icons implementation will be replaced by the useIcon implementation in #21

File: src/components/general/ErrorBoundary/index.tsx (hunk 4 above)
Comment (eskil0312) [reply src/components/general/ErrorBoundary/index.tsx]: Added error and errorMessage on state for now

File: src/components/general/ErrorBoundary/index.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            return message;
+        }
+        if (errorMessage !== "") {
+            return errorMessage;
+        }
+
+        return "Unhandled error message";
+    }
+
+    renderChildren() {
```
Comment (MartinForre) [inline src/components/general/ErrorBoundary/index.tsx]: Is this used by anyone?

File: src/components/general/ErrorBoundary/index.tsx [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        if (this.state.hasError) {
+            return null;
+        }
+
+        return this.props.children;
+    }
+
+    onButtonClick() {
```
Comment (MartinForre) [inline src/components/general/ErrorBoundary/index.tsx]: Sould this be something more along the lines of "takeAction"? E.g takeAction = () => { if(this.state.hasError && this.props.onTakeAction) { return this.props.onTakeAction(); } window.location.reload(); }

The same would apply to the ErrorMessage component. Makes more sense to "Take action" when something goes wrong than to "button" 🙈

File: src/components/general/ErrorBoundary/index.tsx (hunk 5 above)
Comment (eskil0312) [reply src/components/general/ErrorBoundary/index.tsx]: Nope 😄

File: src/components/general/ErrorBoundary/index.tsx (hunk 6 above)
Comment (eskil0312) [reply src/components/general/ErrorBoundary/index.tsx]: Agreed, rewrote method and changed prop name from onButtonClick to onTakeAction

File: src/components/general/ErrorMessage/styles.less [hunk 7]
```diff
@@ -9,7 +9,7 @@
 
     .messageContainer {
         position: relative;
-        top: 0; left: 0; bottom: 0; right: 0;
+        top: 20%; left: 0; bottom: 0; right: 0;
```
Comment (MartinForre) [inline src/components/general/ErrorMessage/styles.less]: Does this have to be percentage?

Comment (eskil0312) [reply src/components/general/ErrorMessage/styles.less]: A but unsure how to solve this. Discussed with Lotte and concluded that the regular error message should always have a 120 px margin from the top. The small message should always be centered. 

Could this be defined in the component like i did in the most recent update? Or should this be left to the user of the component

**Pull request merged**